test/hotspot/jtreg/compiler/conversions/TestMoveConvI2LThroughAddIs.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.conversions;
+
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @bug 8254317
+ * @requires vm.compiler2.enabled
+ * @summary Exercises the optimization that moves integer-to-long conversions
+ *          upwards through different shapes of integer addition
+ *          subgraphs. Contains three basic (small) tests and two stress tests
+ *          that resulted in a compilation time and memory explosion, triggering
+ *          the short specified timeout.
+ * @library /test/lib /
+ * @run main/othervm -Xcomp -XX:-TieredCompilation -XX:-Inline
+ *      -XX:CompileOnly=::testChain,::testTree,::testDAG
+ *      compiler.conversions.TestMoveConvI2LThroughAddIs basic
+ * @run main/othervm/timeout=1 -Xcomp -XX:-TieredCompilation -XX:-Inline
```

Comment:
By default timeout is 120 sec. Why you set it to 1 sec?
The same for second @run.

Reply:
These two runs (`stress1` and `stress2`) test that compilation time does not explode. When it does (before the proposed fix), the runs take > 10s to execute (but less than 120s, at least `stress1`), while after the fix both take just a few ms (all measurements from my local and our CI machines). I just deemed 1s to be a good value to balance the risk of false positives (reporting an nonexistent bug because the timeout is too short) and false negatives (not reporting a real bug because the timeout is too long). I will explore further increasing the load of these runs to see if we can increase the timeout without increasing the risk of false negatives.

Reply:
I have addressed this in the latest update.